src/Foundatio/Caching/CacheClientHelpers.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Foundatio.Caching
+{
+    public static class CacheClientHelpers
+    {
+        public static async Task<CacheValue<T>> GetOrAddAsync<T>(this ICacheClient client, string key, Func<T> addFunc,
+            TimeSpan? expiresIn = null)
```

Comment:
I originally had arg check but then removed it because `GetAsync<T>` does a `String.IsNullOrEmpty(key)`.  I can do the same check prior if you.

I'll take a look at the concurrent collections one, let me get back to that.  I originally got this idea for the GetOrAdd with the `Func<T>` because of looking at another library LazyCache.